tests/tests_e2e/az_ulib_ipc_e2e/az_ulib_ipc_e2e.c

```diff
@@ -657,6 +493,178 @@ static void az_ulib_ipc_e2e_call_sync_command_in_multiple_threads_and_unpublish_
+          "2.123\",\"MY_INTERFACE_3.123\"],\"continuation_token\":655615}")));
+
+  /// cleanup
+  assert_int_equal(az_ulib_ipc_release_interface(query_handle), AZ_OK);
+  unpublish_interfaces_and_deinit_ipc();
+}
+
+static void az_ulib_ipc_query_query_next_succeed(void** state)
+{
+  /// arrange
+  (void)state;
+  init_ipc_and_publish_interfaces(true);
+
+  az_ulib_ipc_interface_handle query_handle;
+  assert_int_equal(
+      az_ulib_ipc_try_get_interface(
+          AZ_SPAN_FROM_STR(IPC_QUERY_1_INTERFACE_NAME),
+          QUERY_1_INTERFACE_VERSION,
+          AZ_ULIB_VERSION_EQUALS_TO,
+          &query_handle),
+      AZ_OK);
+
+  query_1_query_model_in query_in = { .query = AZ_SPAN_FROM_STR("") };
+  uint8_t buf[50];
+
+  /// act
+  /// assert
+  az_span query_result = AZ_SPAN_FROM_BUFFER(buf);
+  query_1_query_model_out query_out = { .result = &query_result, .continuation_token = 0 };
+  assert_int_equal(
+      az_ulib_ipc_call(query_handle, QUERY_1_QUERY_COMMAND, &query_in, &query_out), AZ_OK);
+  assert_true(az_span_is_content_equal(
+      *query_out.result, AZ_SPAN_FROM_STR("\"ipc_query.1\",\"MY_INTERFACE_1.123\"")));
+  assert_int_equal(query_out.continuation_token, 0x000200FF);
+
+  query_1_next_model_in next_in = { .continuation_token = query_out.continuation_token };
+  query_result = AZ_SPAN_FROM_BUFFER(buf);
+  query_1_next_model_out next_out = { .result = &query_result, .continuation_token = 0 };
+  assert_int_equal(
+      az_ulib_ipc_call(query_handle, QUERY_1_NEXT_COMMAND, &next_in, &next_out), AZ_OK);
+  assert_true(az_span_is_content_equal(
+      *next_out.result, AZ_SPAN_FROM_STR("\"MY_INTERFACE_1.2\",\"MY_INTERFACE_2.123\"")));
+  assert_int_equal(next_out.continuation_token, 0x000400FF);
+
+  next_in.continuation_token = next_out.continuation_token;
+  query_result = AZ_SPAN_FROM_BUFFER(buf); // reset az_span size.
+  assert_int_equal(
+      az_ulib_ipc_call(query_handle, QUERY_1_NEXT_COMMAND, &next_in, &next_out), AZ_OK);
+  assert_true(
+      az_span_is_content_equal(*next_out.result, AZ_SPAN_FROM_STR("\"MY_INTERFACE_3.123\"")));
+  assert_int_equal(next_out.continuation_token, 0x000a00FF);
+
+  next_in.continuation_token = next_out.continuation_token;
+  query_result = AZ_SPAN_FROM_BUFFER(buf); // reset az_span size.
+  assert_int_equal(
+      az_ulib_ipc_call(query_handle, QUERY_1_NEXT_COMMAND, &next_in, &next_out), AZ_ULIB_EOF);
+
+  /// cleanup
+  assert_int_equal(az_ulib_ipc_release_interface(query_handle), AZ_OK);
+  unpublish_interfaces_and_deinit_ipc();
+}
+
+static void az_ulib_ipc_query_query_next_w_str_succeed(void** state)
+{
+  /// arrange
+  (void)state;
+  init_ipc_and_publish_interfaces(true);
+
+  az_ulib_ipc_interface_handle query_handle;
+  assert_int_equal(
+      az_ulib_ipc_try_get_interface(
+          AZ_SPAN_FROM_STR(IPC_QUERY_1_INTERFACE_NAME),
+          QUERY_1_INTERFACE_VERSION,
+          AZ_ULIB_VERSION_EQUALS_TO,
+          &query_handle),
+      AZ_OK);
+
+  uint8_t buf[170];
```

Comment:
Can this buf size be a #define? or have a comment describing why it's this size?

Reply:
Better to just add a comment here, this number is important for this paticular test.